### DISCUSSION TITLE
Use of safe-control-gym to validate other control algorithms (as presented in the paper)

### DISCUSSION THREAD

**Discussion by Solomon-GC** [utiasDSL/safe-control-gym#136] [General]
Hello,

Can you please explain how to do the same for GP-MPC?
I followed examples/tutorial.md, but I still have some missing pieces to train the GP-MPC.

Additionally, please orient me on how to add my algorithm.
I would like eventually to test the following: 
1.) Data efficiency as presented in your paper.
2.) Phase plot
3.) Loss

Is it possible to get the reference code to replicate these results from the paper?

Thanks!

**Comment by adamhall** [comment]
Hi!

Sorry, what exactly is meant by "do the same for GP-MPC"?

To see our paper results take a look at the [v1.0.0-alpha](https://github.com/utiasDSL/safe-control-gym/tree/v1.0.0-alpha) branch. There are some instructions on how to reproduce the results there. There can be inconsistencies across different hardware, so if you run into issues, let us know and we can try and help you resolve them. Additionally, a good example of how to use and train GPMPC is given in [here](https://github.com/utiasDSL/safe-control-gym/blob/v1.0.0-alpha/experiments/arxiv/quadrotor_constraint/utils/gpmpc_quadrotor_impossible_traj.py). In the current main branch, we have moved to the `Experiments` class to define the experiments and running the controllers, as you can see in the examples.

To add your own algorithm, you create a child class of [`BaseController`](https://github.com/utiasDSL/safe-control-gym/blob/main/safe_control_gym/controllers/base_controller.py), as well as a YAML file that creates default parameters for your algorithm. You can run it in one of the `Environments` using an `Experiment` class, as done in the examples. 

We have a number of metrics built into the `Experiment` class that can be used to plot your results, including loss, which allows you to construct a data efficiency plot. We don't currently have any built-in way to make a phase plot, but if you end up doing that, we are always looking for contributions.

@Federico-PizarroBejarano @JacopoPan did I miss anything?
